src/Entity/EntityIdValue.php

```diff
@@ -67,7 +60,7 @@ public function unserialize( $serialized ) {
 		try {
 			$entityId = LegacyIdInterpreter::newIdFromTypeAndNumber( $entityType, $numericId );
 		} catch ( InvalidArgumentException $ex ) {
-			throw new IllegalValueException( 'Invalid EntityIdValue serialization.' );
+			throw new IllegalValueException( 'Invalid EntityIdValue serialization.', 0, $ex );
```

Comment:
It's here because it improves the error report when trying to use the LegacyIdInterpreter for custom IDs.